### DISCUSSION TITLE
Can you access the `preview` value from custom _app.js?

### DISCUSSION THREAD

**Discussion by jneterer** [vercel/next.js#21978] [Help] [answered]
I have a NextJS app driven by Sanity CMS. Within Sanity I can manage both the header and footer content which is used for all of my pages. I also have preview mode set up in Sanity to preview all of my Sanity content before deploying it. As of now, on every page I am using `getStaticProps` to get the page data, header data, and footer data separately as you can see in the screenshot below.

`pages/index.tsx`
![Screen Shot 2021-02-08 at 5 22 01 PM](https://user-images.githubusercontent.com/32587255/107288858-29f89c80-6a32-11eb-92e4-cc77f257745d.png)

Ideally, since every page needs the same header and footer data, I would like to move these requests up into `_app.js` as seen in the screenshot below. The issue is if I request the header and footer data in `app.js` in `getInitialProps` I can't figure out how to know whether we are in preview mode or not. Is anyone able to help? Is this even possible? Thanks!

`_app.js`
![Screen Shot 2021-02-08 at 5 27 25 PM](https://user-images.githubusercontent.com/32587255/107289371-ed797080-6a32-11eb-9a6c-ea3627c16f6f.png)

**Comment by ijjk** [comment, marked as answer]
Hi, the `preview` value will be available on the router here once https://github.com/vercel/next.js/pull/21638 lands

**Reply by jneterer** [reply]
@ijjk wow, thanks for the quick response! This is great to hear. Excited for the update.
